### PR TITLE
Lzao/debugger rule removal

### PR DESCRIPTION
**Pull request by zaoliu-aws**
*Issue #, if available:*
This is a draft.

*Description of changes:*

This changes adds a new field DisableProfiler in ProfilerConfig. It also disables the default profiler rules. The changes can be merged after the service package changes are deployed.

*Testing done:*
All the relevant unit/integration tests are updated accordingly to reflect the changes.
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/estimator.py [hunk 1]
```diff
@@ -920,20 +920,25 @@ def _prepare_profiler_for_training(self):
         2. user only specify debugger rules, i.e., rules=[Rule.sagemaker(...)]
         """
         if self.disable_profiler:
-            if self.profiler_config:
+            if self.profiler_config and self.profiler_config.disable_profiler == False:
```
[the remark this answers is not in the document]
Comment (rajanksin) [reply src/sagemaker/estimator.py:920]: I think the you should check for `self.profiler_config.disable_profiler == True` and raise the same error.

[the remark this answers is not in the document]
Comment (zaoliu-aws) [reply src/sagemaker/estimator.py:920]: No. Here self.disable_profiler is the flag from estimator API. when  self.disable_profiler == True, I should check if  self.profiler_config.disable_profiler == False. If it is, it is an invalid configuration and I should rise errors.

**Comment by zaoliu-aws** [comment]
sim ticket for this change.

**Comment by mariumof** [comment]
Looks fine with me.

**Review by knikure — APPROVED** [review]

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Comment by zaoliu-aws** [comment]
AWS SDK changes need to go before the checks can be passed. Otherwise, it will always complaining DisableProfiler flag that is newly introduced. https://trebuchet.corp.amazon.com/manage_feature.html?featureArn=arn:aws:trebuchet:::feature:v2:1ecee220-7512-4e9b-a102-49a0ff52079f

temporarily set to draft.

**Comment by zaoliu-aws** [comment]
AWS SDK changes are released. The issue of unrecognized 'DisableProfiler' should go away.

File: tests/integ/test_profiler.py [hunk 2]
```diff
@@ -423,13 +401,16 @@ def test_mxnet_with_disable_profiler_then_enable_default_profiling(
...
 
         job_description = mx.latest_training_job.describe()
-        assert job_description.get("ProfilerConfig") is None
+        # when the profiler is disabled, ProfilerConfig is not None.
+        # Temporarily remove this check until the service packages are updated.
+        # assert job_description.get("ProfilerConfig") is None
         assert job_description.get("ProfilerRuleConfigurations") is None
-        assert job_description.get("ProfilingStatus") == "Disabled"
+        #  Temporarily remove this check until the service packages are updated.
+        # assert job_description.get("ProfilingStatus") == "Disabled"
```
Comment (mufaddal-rohawala) [inline tests/integ/test_profiler.py:409]: Whats the schedule for service packages to be updated? will we able to make sure the E2E integ test workflow passes before merge.

**Review by mufaddal-rohawala — CHANGES_REQUESTED** [review]

File: tests/integ/test_profiler.py (hunk 2 above)
Comment (zaoliu-aws) [reply tests/integ/test_profiler.py:409]: It is already on the way. I will make sure the E2E integ test workflow passes before merge.

**Review by knikure — COMMENTED** [review]
/bot run pr, notebook-tests, slow-tests

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by navinsoni — COMMENTED** [review]
/bot run pr

**Comment by mariumof** [comment]
Changes for this PR were superseded by https://github.com/aws/sagemaker-python-sdk/pull/3523, which was merged in

**Pull request closed without merging**